compiler/qsc_ast/src/ast.rs

```diff
@@ -1306,6 +1306,122 @@ pub struct Ident {
     pub name: Rc<str>,
 }
 
+/// A [`VecIdent`] represents a sequence of idents. It provides a helpful abstraction
+/// that is more powerful than a simple `Vec<Ident>`, and is primarily used to represent
+/// dot-separated paths.
+#[derive(Clone, Debug, Eq, Hash, PartialEq, Default)]
+pub struct VecIdent(pub Vec<Ident>);
```

Comment:
This and the rest of my `VecIdent` related comments apply to both the AST and HIR versions

Reply:
> I'm curious why it conveys intent better? Lack of dynamic sizing?

Yeah, to me `Box<[Ident]>` gives "immutable array" vibes while `Vec<Ident>` gives "I might add to this later", since one of the core attributes of `Vec` is that it "capacity" and "length" are decoupled and over-allocating is the default.